src/client/pfs_v2.go

```diff
@@ -267,3 +292,31 @@ func (r *getTarConditionalReader) drain() error {
 		}
 	}
 }
+
+func (c APIClient) CreateTempFileSet(r io.Reader, tag ...string) (string, error) {
+	ctx := c.Ctx()
+	client, err := c.PfsAPIClient.CreateTempFileSet(ctx)
+	if err != nil {
+		return "", err
+	}
+	foc := FileOperationClient{
+		client: client,
+	}
+	if err := foc.PutTar(r, tag...); err != nil {
+		return "", err
+	}
+	res, err := foc.closeRecvTempFileSet()
+	if err != nil {
+		return "", err
+	}
+	return res.FilesetId, nil
+}
+
+const TempRepoName = "__tmp__"
```

Comment:
Since we have a separate endpoint, I don't think this is necessary on the client side. Also, it looks like the below function is unused.

Reply:
This is used to construct the commit for the File methods. Instead of having a separate set of functions for commits, we have a sentinel repo that causes the commit id to be interpreted as a temp fileset id.

Reply:
`TempRepoName` -> `TmpRepoName`